app/models/repo.rb

```diff
@@ -389,6 +389,20 @@ def set_image_data
     self.save!
   end
 
+  # Returning MetadataSource that contains structural metadata.
+  def structural_metadata
+    metadata_builder.metadata_source.find do |metadata|
```

Comment:
i'd suggest:

```
metadata_builder.metadata_source.find_by source_type: MetadataSource::STRUCTURAL_TYPES
```

Reply:
ooooo that's way better. I'll change it, thanks!